### PR TITLE
Update Tasks

### LINKED ISSUE #1: CRUD Capability
Add a way to delete and edit tasks.

### PR DESCRIPTION
**Pull request by markdstouffer**
## Update
- Double clicking on any task will now replace that task with an input element, in which the user can modify the task
- Finish editing by clicking the `Escape` or `Enter` keys
- Added editNote function to the store service (noteService)
- Resolves #1 

### Note
- Ignore duplicate commits, not sure why my fork seems to be messed up!

### CODE REVIEW

File: src/components/DailyChecklist.global.css [hunk 1]
```diff
@@ -43,7 +42,29 @@ input[type='checkbox']:checked + label {
...
 #weekday_date {
   padding: 0px;
-  margin: 0px;
-  text-align: center;
+  margin-left: 20px;
+  text-align: left;
   font-size: 19px;
 }
+
+#more {
```
Comment (cakoshakib) [inline src/components/DailyChecklist.global.css:50]: rename to something more descriptive

File: src/components/DailyChecklist.tsx [hunk 2]
```diff
@@ -19,25 +28,49 @@ const Checkbox = ({ task }: { task: string }) => {
...
+      ) : (
+        <input
+          className="task_text"
+          type="text"
+          value={val}
+          onChange={(event) => {
+            const content = event.target.value;
+            noteService.editNote(day, index, content);
+            setVal(content);
+          }}
```
Comment (cakoshakib) [inline src/components/DailyChecklist.tsx:48]: Style this input field if you can to be smaller and not the same as the field to add tasks

Just need to add a unique id or class to the editing input and creation input

File: src/components/DailyChecklist.tsx [hunk 3]
```diff
@@ -48,27 +81,22 @@ const Tasks = ({ day }: Props) => {
...
+      <div id="tasks">
         {tasks.map((task, index) => (
-          <div key={uuidv4()}>
-            <Checkbox key={uuidv4()} task={task} />
-            <button type="submit" onClick={() => handleDelete(index)}>
-              delete
-            </button>
+          <div key={uuidv4()} id="editing">
+            <Checkbox key={uuidv4()} task={task} index={index} day={day} />
+            <MdClose
```
Comment (cakoshakib) [inline src/components/DailyChecklist.tsx:95]: Not an issue, but what does this do?

File: src/components/DailyChecklist.tsx [hunk 4]
```diff
@@ -78,12 +106,11 @@ const Tasks = ({ day }: Props) => {
...
   return (
-    <div>
-      <button type="submit" onClick={handleClick}>
-        edit
-      </button>
-      {tasks.map((task) => (
-        <Checkbox key={uuidv4()} task={task} />
+    <div id="tasks">
+      {tasks.map((task, index) => (
+        <div key={uuidv4()} id="editing">
```
Comment (cakoshakib) [inline src/components/DailyChecklist.tsx:111]: I feel like the id "editing" doesn't really make sense in this context

File: src/services/notes.tsx [hunk 5]
```diff
@@ -22,4 +22,10 @@ const deleteNote = (day: string, index: number) => {
   store.set(day, dayArr);
 };
 
-export default { getDay, addNote, deleteNote };
+const editNote = (day: string, index: number, newNote: string) => {
+  const dayArr = store.get(day) as string[];
+  dayArr[index] = newNote;
+  store.set(day, dayArr);
+};
```
Comment (cakoshakib) [inline src/services/notes.tsx:29]: This is clean 🧊

**Review by cakoshakib — COMMENTED** [review]
Functionality is perfect, works exactly as I would have done it myself. Just a few stylistic changes mentioned.

File: src/components/DailyChecklist.tsx (hunk 3 above)
Comment (markdstouffer) [reply src/components/DailyChecklist.tsx:95]: Delete buttons (previous PR)

File: src/components/DailyChecklist.tsx (hunk 4 above)
Comment (markdstouffer) [reply src/components/DailyChecklist.tsx]: fdd48ba

File: src/components/DailyChecklist.tsx (hunk 2 above)
Comment (markdstouffer) [reply src/components/DailyChecklist.tsx:48]: fdd48ba

File: src/components/DailyChecklist.global.css (hunk 1 above)
Comment (markdstouffer) [reply src/components/DailyChecklist.global.css]: fdd48ba

**Pull request merged**